libminifi/src/io/FileStream.cpp

```diff
@@ -28,12 +28,15 @@ namespace nifi {
 namespace minifi {
 namespace io {
 
-FileStream::FileStream(const std::string &path)
+FileStream::FileStream(const std::string &path, bool append)
     : logger_(logging::LoggerFactory<FileStream>::getLogger()),
       path_(path),
       offset_(0) {
   file_stream_ = std::unique_ptr<std::fstream>(new std::fstream());
-  file_stream_->open(path.c_str(), std::fstream::out | std::fstream::binary);
+  if (append)
+    file_stream_->open(path.c_str(), std::fstream::in | std::fstream::out | std::fstream::app | std::fstream::binary);
```

Comment:
Why do we need "in" for appending? 

To avoid duplication can it be done like:
```
auto flags  = std::fstream::out | std::fstream::binary;
if (append) {
  flags |= std::fstream::in | std::fstream::app
}
file_stream_->open(path.c_str(), flags);
```
?

Reply:
That bit sets the flag to allow output operations to append at the end but there is nothing to suggest we won't be reading. Further, we need to be able to read and seek to get length. We don't want to obviate user's intent.